### ISSUE TITLE
Using orderBy in ng-repeat causes item to duplicate and JSON to not update properly

### ISSUE THREAD

**Issue by srimola** [marceljuenemann/angular-drag-and-drop-lists#142]
If I use an `orderBy` in `<li ng-repeat>`, it overrides the view with duplicate items, and the JSON does not update properly.

**Comment by damdoum** [comment]
Hi @srimola ,
Did you try to " track by " something the elements in your ng-repeat. I have met that problem before and i resolve it like that. So give a try to something like that:

    ng-repeat="item in items | orderBy:item.name track by $index"

 for example... 
I think that it shouldn't be related to that module. If it will resolve your issue or if you have already resolved it please give me feed back i am planning to use  <b>angular-drag-and-drop-lists</b> too

**Comment by sumvee** [comment]
OrderBy messes things up ! even on using track by $index or  track by module.id

**Comment by jjoedouglas** [comment]
I've been able to use order by.  Can you provide a plunker or gist displaying the problem?

**Comment by marceljuenemann** [comment]
Using this directive with OrderBy doesn't really make sense to me. It means the user can't determine the position of the element in the list, so why allow reordering?

**Comment by dbounds** [comment]
@marceljuenemann, Reordering is a specific feature of drag/drop lists which shouldn't be a requirement. Many are using this directive not for the utility of re-ordering but rather just for drag/drop.

**Comment by DabeDotCom** [comment]
Sorry to drag up an old topic, but I just wanted to mention that I recently ran into the same problem:

``` html
<!DOCTYPE html>
<html><head><meta charset="utf-8"/><style>
body > div   { width:33%; margin:1em auto; font-size:22px }
ul[dnd-list] { padding-left:0px; margin:0 }
ul[dnd-list] .dndDraggingSource { display:none }
ul[dnd-list] .dndPlaceholder    { background-color:#ddd }
ul[dnd-list] li { display:block; border:1px solid #ddd; margin-bottom:-1px; min-height:24px; padding:4px }
</style></head>

<body ng-app="App">
<div ng-controller="DragCtrl">

  <h4 style="margin:0">Items: {{ list.length }}</h4>
  <ul dnd-list="list">
    <!-- This might have more complex "orderBy" logic (e.g., more columns) -->
    <li ng-repeat="item in list |orderBy:['num']"
        dnd-draggable="item"
        dnd-moved="list.splice($index, 1)"
        dnd-effect-allowed="move">
        {{item.label}}
    </li>
  </ul>
</div>

<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/angular-drag-and-drop-lists/1.4.0/angular-drag-and-drop-lists.min.js"></script>

<script type="text/javascript">
angular.module('App', ['dndLists'])

.controller('DragCtrl', ['$scope', '$log', function($scope, $log) {
   $scope.list = [
      // These might arrive out-of-order from an AJAX service, etc.
      { label: "To Here", num: 34567 },
      { label: "Drag Me", num: 23456 }
   ];
}]);
</script>
</body></html>
```

My intention was to use `|orderBy` to specify the *initial* sorting, but then be able to shuffle things around **after** that...

FWIW, my eventual solution/workaround was to manually sort the list in my controller, instead:

```
  ItemResource.get(function(result) {
    result.sort(function(a, b) {
      if ( a.group_id < b.group_id ) { return -1 }
      if ( a.group_id > b.group_id ) { return  1 }
      if ( a.year < b.year ) { return -1 }
      if ( a.year > b.year ) { return  1 }
      if ( a.last_name > b.last_name ) { return -1 }
      if ( a.last_name < b.last_name ) { return  1 }
      if ( a.first_name < b.first_name ) { return -1 }
      if ( a.first_name > b.first_name ) { return  1 }
      // else
      return 0;
    });
    for (var i=0; i<result.length; i++) {
      push($scope.list, {
        label: result[i].first_name + " " + result[i].last_name,
         item: result[i]
      });
    }
  });
```

I can't say that I love that, though...  [For instance, did you spot the typo?]
